framework/src/problems/EigenProblem.C

```diff
@@ -517,7 +523,11 @@ EigenProblem::solve()
       {
         _console << " Free power iteration starts" << std::endl;
         doFreeNonlinearPowerIterations(solverParams()._free_power_iterations);
-        _first_solve = false;
+
+        // If there is a source term in the nonlinear problem, the scaling
+        // on the initial condition for ||Bx||_2 = inverse of the eigenvalue
+        // is no longer valid so we will need to re-scale again
+        preScaleEigenVector();
```

Comment:
Power iteration is solving Ax = Bx^old / ||Bx^old||_2, and k = ||Bx||_2, so `preScaleEigenVector()` will be a null operation here. Code wise, `_nl_eigen->getConvergedEigenvalue(_active_eigen_index)` will just return ||Bx||_2, so the scaling factor will be one.

Reply:
Like
```
      if (solverParams()._free_power_iterations)
      {
       MooseDoOnce(
        _console << " Free power iteration starts" << std::endl;
        doFreeNonlinearPowerIterations(solverParams()._free_power_iterations);
        );
      }
```